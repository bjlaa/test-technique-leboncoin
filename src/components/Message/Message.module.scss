.message {
	background: red;
	border-radius: 10px;
	padding: 10px;
	width: 70%;
	position: relative;
	font-size: 10pt;
	margin: 10px 0;
	text-align: left;
	&:nth-child(even) {
		margin-left: auto;
		background-color: rgba(255, 100, 10, 1);
		color: white;
		&:after {
			content: ' ';
			position: absolute;
			bottom: 0;
			left: 82%;
			width: 0;
			height: 0;
			border: 10px solid transparent;
			border-top-color: rgba(255, 100, 10, 1);
			border-bottom: 0;
			border-right: 0;
			margin-left: -10px;
			margin-bottom: -10px;
		}
	}
	&:nth-child(odd) {
		margin-right: auto;
		background-color: rgba(0, 170, 187, 0.3);
		&:after {
			content: ' ';
			position: absolute;
			bottom: 0;
			left: 24%;
			width: 0;
			height: 0;
			border: 10px solid transparent;
			border-top-color: rgba(0, 170, 187, 0.3);
			border-bottom: 0;
			border-left: 0;
			margin-left: -10px;
			margin-bottom: -10px;
		}
	}
	&__private {
		font-size: 8pt;
		margin-top: 10px;
	} 
}