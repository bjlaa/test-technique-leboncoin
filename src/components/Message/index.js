import React from 'react';

import styles from './Message.module.scss';

const Message = ({ message }) => {
	if (!message) {
		return null;
	}

	return (
		<div className={styles.message}>
			{
				message.private
				?
					<div className={`${styles.message__private} message__private`}>This message was made private by the user.</div>
				:
					<span>{message.text}</span>
			}
		</div>
	);
}

export default Message;
