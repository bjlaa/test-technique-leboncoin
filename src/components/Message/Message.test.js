import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Message from './index';

configure({ adapter: new Adapter() });

describe('Message', () => {
	const messageText = 'test';
	const message = { id: 0, text: messageText };

	it('should match its snapshot', () => {
		const component = renderer.create(<Message />);

		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should show the message passed via the message prop', () => {
		const component = shallow(<Message message={message} />);

		expect(component.find('span').text()).toEqual(messageText);
	});

	it('should should show a div bearing the private id if the message is private', () => {
		message.private = true;

		const component = shallow(<Message message={message} />);

		expect(component.find('.message__private').length).toEqual(1);
	});
});
