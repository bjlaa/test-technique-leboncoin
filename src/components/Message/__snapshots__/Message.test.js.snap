// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match its snapshot 1`] = `null`;
